engine/table/src/main/java/io/deephaven/engine/util/BigDecimalUtils.java

```diff
@@ -44,67 +52,148 @@ public PrecisionAndScale(final int precision, final int scale) {
     /**
      * Compute an overall precision and scale that would fit all existing values in a table.
      *
-     * @param t a Deephaven table
+     * @param table a Deephaven table
      * @param colName a Column for {@code t}, which should be of {@code BigDecimal} type
      * @return a {@code PrecisionAndScale} object result.
      */
     public static PrecisionAndScale computePrecisionAndScale(
-            final Table t, final String colName) {
-        final ColumnSource<BigDecimal> src = t.getColumnSource(colName, BigDecimal.class);
-        return computePrecisionAndScale(t.getRowSet(), src);
+            final Table table, final String colName) {
+        final ColumnSource<BigDecimal> src = table.getColumnSource(colName, BigDecimal.class);
```

Comment:
Since you validate type/component type later, you can drop the type assertion and simplify. Otherwise, this method is only useful with type BD, not component type BD.
```suggestion
        final ColumnSource<?> src = table.getColumnSource(colName);
```